src/main/java/by/custom/utilcalculator/domain/FileUserProgressStorage.java

```diff
@@ -0,0 +1,60 @@
+package by.custom.utilcalculator.domain;
+
+import by.custom.utilcalculator.exception.UserFileNotFoundException;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FileUserProgressStorage implements iUserProgressStorage {
+    private final String path = "D:/%s.dat";
+
+    private static class FileUserProgressStorageHolder {
+        private static final FileUserProgressStorage FILE_USER_PROGRESS_STORAGE = new FileUserProgressStorage();
+    }
+
+    private FileUserProgressStorage() {
+    }
+
+    public static FileUserProgressStorage getInstance() {
+        return FileUserProgressStorageHolder.FILE_USER_PROGRESS_STORAGE;
+    }
+
+    @Override
+    public void save(UserProgress userProgress) {
+        try {
+            ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(String.format(path, userProgress.getChatID())));
```

Comment:
What do you think to encapsulate building path to file logic in dedicated method?
```suggestion
            ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(buildFilePath(path, userProgress.getChatID())));
```

Reply:
Good point. Done.